File: 232. Implement Queue using Stacks.java
import java.util.Stack;

class MyQueue {

    Stack<Integer> s1 = new Stack<>();
    Stack<Integer> s2 = new Stack<>();


    public MyQueue() { }

    public void push(int x) {
        s1.push(x);
    }

    private void transferStackData(){
        while(!s1.isEmpty()){
            s2.push(s1.peek());
            s1.pop();
        }
    }

    public int pop() {

        if(!s2.isEmpty()){
            Integer value = s2.peek();
            s2.pop();
            return value;
        }

        transferStackData();
        Integer value = s2.peek();
        s2.pop();

        return value;

    }

    public int peek() {
        if(!s2.isEmpty()){
            return s2.peek();
        }

        transferStackData();

        return s2.peek();

    }

    public boolean empty() {
        return s1.isEmpty() && s2.isEmpty();
    }
}

File: 232. Implement Queue using Stacks.py
class MyQueue:
    def __init__(self): self.s1, self.s2 = collections.deque(), collections.deque()
    def push(self, x: int) -> None: self.s1.append(x)
    def s1Tos2(self,remove: bool) -> int:        
        while len(self.s1) > 1: self.s2.append(self.s1.pop())
        val = self.s1.pop()
        if not remove: self.s2.append(val)
        while self.s2: self.s1.append(self.s2.pop())
        return val
    def pop(self) -> int: return self.s1Tos2(True)
    def peek(self) -> int: return self.s1Tos2(False)
    def empty(self) -> bool: return len(self.s1) == 0
        
        


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()